### PR TITLE
[3/4] refactoring04: EntityViewController

### PR DESCRIPTION
**Pull request by nickAS21**
## Pull Request description

Put your PR description here instead of this sentence.   

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] PR name contains fix version. For example, "[3.3.4] Hotfix of some UI component" or "[3.4] New Super Feature".
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Required for internal contributors only.

## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.

### CODE REVIEW

File: application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.thingsboard.server.dao.service.Validator.validateId;
+
+@Service
+@TbCoreComponent
+@AllArgsConstructor
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java:64]: You do not use this annotations.
Do TimeseriesService final or remove this annotation

File: application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+            notificationEntityService.notifyAssignOrUnassignEntityToCustomer(tenantId, entityView.getId(), customer.getId(), savedEntityView,
+                    actionType, EdgeEventActionType.UNASSIGNED_FROM_CUSTOMER, user, true, customer.getId().toString(), customer.getName());
+            return savedEntityView;
+        } catch (Exception e) {
+            notificationEntityService.notifyEntity(tenantId, emptyId(EntityType.ENTITY_VIEW), null, null,
+                    actionType, user, e, entityView.getId().toString());
+            throw handleException(e);
+        }
+    }
+
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java:224]: remove extra line

File: application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+                        log.error("Failed to log timeseries delete", e);
+                    }
+                    resultFuture.setException(t);
+                }
+            });
+        }
+        return resultFuture;
+    }
+
+    private void logAttributesUpdated(SecurityUser user, EntityId entityId, String scope, List<AttributeKvEntry> attributes, Throwable e) throws ThingsboardException {
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java:391]: move these log methods to notification service

File: application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+                    edgeId.toString(), edge.getName());
+            return savedDevice;
+        } catch (Exception e) {
+            notificationEntityService.notifyEntity(tenantId, emptyId(EntityType.ENTITY_VIEW), null, null,
+                    actionType, user, e, entityViewId.toString(), edgeId.toString());
+            throw handleException(e);
+        }
+    }
+
+    //
```
[the remark this answers is not in the document]
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java:208]: OK

File: application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java (hunk 2 above)
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java:224]: Ok

File: application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java [hunk 5]
```diff
@@ -0,0 +1,418 @@
...
+
+                    @Override
+                    public void onFailure(Throwable t) {
+                    }
+                });
+            }
+            return null;
+        }, MoreExecutors.directExecutor());
+    }
+
```
[the remark this answers is not in the document]
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java:307]: Ok

File: application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java [hunk 6]
```diff
@@ -389,15 +384,15 @@ public void onFailure(Throwable t) {
     }
 
     private void logAttributesUpdated(SecurityUser user, EntityId entityId, String scope, List<AttributeKvEntry> attributes, Throwable e) throws ThingsboardException {
-        logEntityAction(user, user.getTenantId(), entityId, null, null, ActionType.ATTRIBUTES_UPDATED, toException(e), scope, attributes);
+        notificationEntityService.notifyEntity(user.getTenantId(), entityId, null, null, ActionType.ATTRIBUTES_UPDATED, user, toException(e), scope, attributes);
     }
 
     private void logAttributesDeleted(SecurityUser user, EntityId entityId, String scope, List<String> keys, Throwable e) throws ThingsboardException {
-        logEntityAction(user, user.getTenantId(), entityId, null, null, ActionType.ATTRIBUTES_DELETED, toException(e), scope, keys);
+        notificationEntityService.notifyEntity(user.getTenantId(), entityId, null, null, ActionType.ATTRIBUTES_DELETED, user, toException(e), scope, keys);
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java:391]: NPE

File: application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java [hunk 7]
```diff
@@ -389,15 +384,15 @@ public void onFailure(Throwable t) {
...
     }
 
     private void logTimeseriesDeleted(SecurityUser user, EntityId entityId, List<String> keys, Throwable e) throws ThingsboardException {
-        logEntityAction(user, user.getTenantId(), entityId, null, null, ActionType.TIMESERIES_DELETED, toException(e), keys);
+        notificationEntityService.notifyEntity(user.getTenantId(), entityId, null, null, ActionType.TIMESERIES_DELETED, user, toException(e), keys);
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java:395]: NPE

File: application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java [hunk 8]
```diff
@@ -0,0 +1,410 @@
...
+
+    private void logAttributesDeleted(SecurityUser user, EntityId entityId, String scope, List<String> keys, Throwable e) throws ThingsboardException {
+        notificationEntityService.notifyEntity(user.getTenantId(), entityId, null, null, ActionType.ATTRIBUTES_DELETED, user, toException(e), scope, keys);
+    }
+
+    private void logTimeseriesDeleted(SecurityUser user, EntityId entityId, List<String> keys, Throwable e) throws ThingsboardException {
+        notificationEntityService.notifyEntity(user.getTenantId(), entityId, null, null, ActionType.TIMESERIES_DELETED, user, toException(e), keys);
+    }
+
+    private EntityView checkEntityViewId(EntityViewId entityViewId, Operation operation, SecurityUser user) throws ThingsboardException {
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java:395]: remove checkEntityViewId
We do id check in controller layer.

Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java]: OK

File: application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java (hunk 7 above)
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java:379]: ok

File: application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java (hunk 6 above)
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java:375]: ok

File: application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java [hunk 9]
```diff
@@ -389,15 +384,15 @@ public void onFailure(Throwable t) {
     }
 
     private void logAttributesUpdated(SecurityUser user, EntityId entityId, String scope, List<AttributeKvEntry> attributes, Throwable e) throws ThingsboardException {
-        logEntityAction(user, user.getTenantId(), entityId, null, null, ActionType.ATTRIBUTES_UPDATED, toException(e), scope, attributes);
+        notificationEntityService.notifyEntity(user.getTenantId(), entityId, null, null, ActionType.ATTRIBUTES_UPDATED, user, toException(e), scope, attributes);
```
[the remark this answers is not in the document]
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/service/entitiy/entityView/DefaultTbEntityViewService.java:371]: ok

File: application/src/main/java/org/thingsboard/server/controller/EntityViewController.java [hunk 10]
```diff
@@ -775,27 +470,16 @@ public EntityView unassignEntityViewFromEdge(@PathVariable(EDGE_ID) String strEd
...
 
-            EntityView savedEntityView = checkNotNull(entityViewService.unassignEntityViewFromEdge(getTenantId(), entityViewId, edgeId));
-            logEntityAction(entityViewId, entityView,
-                    entityView.getCustomerId(),
-                    ActionType.UNASSIGNED_FROM_EDGE, null, strEntityViewId, strEdgeId, edge.getName());
+        EntityViewId entityViewId = new EntityViewId(toUUID(strEntityViewId));
+        EntityView entityView = checkEntityViewId(entityViewId, Operation.READ);
 
-            sendEntityAssignToEdgeNotificationMsg(getTenantId(), edgeId, savedEntityView.getId(), EdgeEventActionType.UNASSIGNED_FROM_EDGE);
+        EntityView savedEntityView = checkNotNull(entityViewService.unassignEntityViewFromEdge(getTenantId(), entityViewId, edgeId));
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:480]: do this in TbEntityViewService

Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:480]: do this in TbEntityViewService

File: application/src/main/java/org/thingsboard/server/controller/EntityViewController.java [hunk 11]
```diff
@@ -738,27 +446,14 @@ public EntityView assignEntityViewToEdge(@PathVariable(EDGE_ID) String strEdgeId
...
-            return savedEntityView;
-        } catch (Exception e) {
-            logEntityAction(emptyId(EntityType.ENTITY_VIEW), null,
-                    null,
-                    ActionType.ASSIGNED_TO_EDGE, e, strEntityViewId, strEdgeId);
-            throw handleException(e);
-        }
+        EntityViewId entityViewId = new EntityViewId(toUUID(strEntityViewId));
+        checkEntityViewId(entityViewId, Operation.READ);
+        EntityView savedEntityView = checkNotNull(entityViewService.assignEntityViewToEdge(getTenantId(), entityViewId, edgeId));
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:455]: do this in TbEntityViewService

Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:455]: do this in TbEntityViewService

File: application/src/main/java/org/thingsboard/server/controller/EntityViewController.java [hunk 12]
```diff
@@ -705,23 +423,13 @@ public EntityView assignEntityViewToPublicCustomer(
...
-        } catch (Exception e) {
-            logEntityAction(emptyId(EntityType.ENTITY_VIEW), null,
-                    null,
-                    ActionType.ASSIGNED_TO_CUSTOMER, e, strEntityViewId);
-            throw handleException(e);
-        }
+        EntityViewId entityViewId = new EntityViewId(toUUID(strEntityViewId));
+        checkEntityViewId(entityViewId, Operation.ASSIGN_TO_CUSTOMER);
+        Customer publicCustomer = customerService.findOrCreatePublicCustomer(getTenantId());
+        EntityView savedEntityView = checkNotNull(entityViewService.assignEntityViewToCustomer(getTenantId(),
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:429]: do this in TbEntityViewService

Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:429]: do this in TbEntityViewService

File: application/src/main/java/org/thingsboard/server/controller/EntityViewController.java [hunk 13]
```diff
@@ -484,28 +215,15 @@ public EntityView unassignEntityViewFromCustomer(
...
-                    ActionType.UNASSIGNED_FROM_CUSTOMER, e, strEntityViewId);
-            throw handleException(e);
+        EntityViewId entityViewId = new EntityViewId(toUUID(strEntityViewId));
+        EntityView entityView = checkEntityViewId(entityViewId, Operation.UNASSIGN_FROM_CUSTOMER);
+        if (entityView.getCustomerId() == null || entityView.getCustomerId().getId().equals(ModelConstants.NULL_UUID)) {
+            throw new IncorrectParameterException("Entity View isn't assigned to any customer!");
         }
+
+        Customer customer = checkCustomerId(entityView.getCustomerId(), Operation.READ);
+        EntityView savedEntityView = checkNotNull(entityViewService.unassignEntityViewFromCustomer(getTenantId(), entityView.getId()));
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:225]: do this in TbEntityViewService

Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:225]: do this in TbEntityViewService

File: application/src/main/java/org/thingsboard/server/controller/EntityViewController.java [hunk 14]
```diff
@@ -451,28 +195,15 @@ public EntityView assignEntityViewToCustomer(
...
+        checkEntityViewId(entityViewId, Operation.ASSIGN_TO_CUSTOMER);
 
-            return savedEntityView;
-        } catch (Exception e) {
-            logEntityAction(emptyId(EntityType.ENTITY_VIEW), null,
-                    null,
-                    ActionType.ASSIGNED_TO_CUSTOMER, e, strEntityViewId, strCustomerId);
-            throw handleException(e);
-        }
+        EntityView savedEntityView = checkNotNull(entityViewService.assignEntityViewToCustomer(getTenantId(), entityViewId, customerId));
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:205]: do this in TbEntityViewService

Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:205]: do this in TbEntityViewService

File: application/src/main/java/org/thingsboard/server/controller/EntityViewController.java (hunk 10 above)
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:480]: Ok

Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:480]: Ok

File: application/src/main/java/org/thingsboard/server/controller/EntityViewController.java (hunk 11 above)
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:455]: Ok

Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:455]: Ok

File: application/src/main/java/org/thingsboard/server/controller/EntityViewController.java (hunk 12 above)
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:429]: Ok

Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:429]: Ok

File: application/src/main/java/org/thingsboard/server/controller/EntityViewController.java (hunk 13 above)
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/controller/EntityViewController.java:225]: Ok

**Pull request closed without merging**